identity-service/src/routes/healthCheck.js

```diff
@@ -76,29 +95,70 @@ module.exports = function (app) {
       if (block && block.transactions.length) {
         for (const tx of block.transactions) {
           // If transaction is from audius account, determine success or fail status
           if (RELAY_HEALTH_ACCOUNT === tx.from) {
             const txHash = tx.hash
             const resp = await web3.eth.getTransactionReceipt(txHash)
             txCounter++
-            if (!resp.status) failureTxHashes.push(txHash)
+
+            // tx failed
+            if (!resp.status) {
+              const senderAddress = await redis.hget('txHasToSenderAddress', txHash)
+              if (senderAddress) {
+                if (!failureTxs[senderAddress]) failureTxs[senderAddress] = [txHash]
+                else failureTxs[senderAddress].push(txHash)
+              } else failureTxs['unknown'].push(txHash)
+            }
           }
         }
       }
     }
 
+    let isError = false
+    if (Object.keys(failureTxs).length >= RELAY_HEALTH_MIN_NUM_USERS &&
+      flatten(Object.values(failureTxs)).length > maxErrors) isError = true
+
+    // delete old entries from set in redis
+    const epochOneHourAgo = Math.floor(Date.now() / 1000) - 3600
+    await redis.zremrangebyscore('relayTxAttempts', '-inf', epochOneHourAgo)
+    await redis.zremrangebyscore('relayTxFailures', '-inf', epochOneHourAgo)
+    await redis.zremrangebyscore('relayTxSuccesses', '-inf', epochOneHourAgo)
+
+    // check if there have been any attempts in the time window that we processed the block health check
+    const attemptedTxsInRedis = await redis.zrangebyscore('relayTxAttempts', minBlockTime, '+inf')
+    const successfulTxsInRedis = await redis.zrangebyscore('relayTxSuccesses', minBlockTime, '+inf')
+    const failureTxsInRedis = await redis.zrangebyscore('relayTxFailures', minBlockTime, '+inf')
+    if ((txCounter / attemptedTxsInRedis.length) < sentVsAttemptThreshold) isError = true
+
     const serverResponse = {
-      numberOfTransactions: txCounter,
-      numberOfFailedTransactions: failureTxHashes.length,
-      failedTransactionHashes: failureTxHashes,
-      startBlock: startBlockNumber,
-      endBlock: endBlockNumber,
+      blockchain: {
+        numberOfTransactions: txCounter,
+        numberOfFailedTransactions: flatten(Object.values(failureTxs)).length,
+        failedTransactionHashes: failureTxs,
+        startBlock: startBlockNumber,
+        endBlock: endBlockNumber
+      },
+      redis: {
+        attemptedTxsCount: attemptedTxsInRedis.length,
+        successfulTxsCount: successfulTxsInRedis.length,
+        failureTxsCount: failureTxsInRedis.length
+      },
       timeElapsed: Date.now() - start
     }
 
-    if (failureTxHashes.length > maxErrors) return errorResponseServerError(serverResponse)
+    if (isVerbose) {
+      serverResponse.redis = {
+        ...serverResponse.redis,
+        attemptedTxsInRedis,
+        successfulTxsInRedis,
+        failureTxsInRedis
+      }
+    }
```

Comment:
nit but maybe blockchain.failedTransactionHashes should also be verbose conditional?

Reply:
could you elaborate on this?

Reply:
discussed offline